base/loading.jl

```diff
@@ -525,7 +527,23 @@ function pkgdir(m::Module, paths::String...)
     rootmodule = moduleroot(m)
     path = pathof(rootmodule)
     path === nothing && return nothing
-    return joinpath(dirname(dirname(path)), paths...)
+    original = path
+    path = dirname(path)
+    if endswith(path, "src")
+        path = dirname(path)
+    elseif contains(path, "ext")
+        # extensions can reside at `../ext/FooExt.jl` or `../ext/FooExt/FooExt.jl`
+        if endswith(path, "ext")
```

Comment:
Would this be correct if `path` is something like `../test/QuickText`?
(since `endswith("QuickText", "ext") === true`)

Reply:
Indeed. Thanks